examples/infection_phenotyping/Infection_classification_model.py

```diff
@@ -0,0 +1,251 @@
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        lr: float = 1e-3,
+        loss_function: nn.CrossEntropyLoss = None,
+        schedule: Literal["WarmupCosine", "Constant"] = "Constant",
+        log_batches_per_epoch: int = 2,
+        log_samples_per_batch: int = 1,
+    ):
+        super(LightningUNet, self).__init__()
+        self.unet_model = Unet2d(in_channels=in_channels, out_channels=out_channels)
+        self.lr = lr
+        self.loss_function = loss_function if loss_function else nn.CrossEntropyLoss()
+        self.schedule = schedule
+        self.log_batches_per_epoch = log_batches_per_epoch
+        self.log_samples_per_batch = log_samples_per_batch
+        self.training_step_outputs = []
+        self.validation_step_outputs = []
+
+    def forward(self, x):
+        return self.unet_model(x)
+
+    def configure_optimizers(self):
+        optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
+        return optimizer
+
+    def training_step(self, batch: Sample, batch_idx: int):
+
+        # Extract the input and target from the batch
+        source = batch["source"]
+        target = batch["target"]
+        pred = self.forward(source)
+
+        # Convert the target image to one-hot encoding
+        target_one_hot = F.one_hot(target.squeeze(1).long(), num_classes=4).permute(
+            0, 4, 1, 2, 3
+        )
+        target_one_hot = target_one_hot.float()  # Convert target to float type
+        # Calculate the loss
+        train_loss = self.loss_function(pred, target_one_hot)
+        if batch_idx < self.log_batches_per_epoch:
+            self.training_step_outputs.extend(
+                self._detach_sample((source, target_one_hot, pred))
+            )
+        self.log(
+            "loss/train",
+            train_loss,
+            on_step=True,
+            on_epoch=True,
+            prog_bar=True,
+            logger=True,
+            sync_dist=True,
+        )
+        return train_loss
+
+    def validation_step(self, batch: Sample, batch_idx: int):
+
+        # Extract the input and target from the batch
+        source = batch["source"]
+        target = batch["target"]
+        pred = self.forward(source)
+
+        # Convert the target image to one-hot encoding
+        target_one_hot = F.one_hot(target.squeeze(1).long(), num_classes=4).permute(
+            0, 4, 1, 2, 3
+        )
+        target_one_hot = target_one_hot.float()  # Convert target to float type
+        # Calculate the loss
+        loss = self.loss_function(pred, target_one_hot)
+        if batch_idx < self.log_batches_per_epoch:
+            self.validation_step_outputs.extend(
+                self._detach_sample((source, target_one_hot, pred))
+            )
+        self.log(
+            "loss/validate",
+            loss,
+            sync_dist=True,
+            add_dataloader_idx=False,
+            logger=True,
+        )
+        return loss
+
```

Comment:
def on_validation_epoch_end(self):
        # Log the confusion matrix at the end of the epoch
        confusion_matrix = self.val_cm.compute().cpu().numpy()

        self.logger.experiment.add_figure(
            "Validation Confusion Matrix",
            plot_confusion_matrix(confusion_matrix, self.index_to_label),
            self.current_epoch,
        )
        self.val_cm.reset()